### ISSUE TITLE
OSX startup crashed on loading channels/PMs

### ISSUE THREAD

**Issue by jottinger** [eul-im/eul#96]
On initial app startup, Volt started displaying the various active channels and active user chats, then crashed. Subsequent startup was slow but working.

**Comment by voltapp2** [comment]
Hi @jottinger 

Is the Slack workspace you are testing public?

**Comment by barberousse** [comment]
Same issue. Mine is a company-owned Slack.

**Comment by AdamVig** [comment]
Same issue, but after starting Volt and authenticating again (which is a pain), it crashed again.

At the very least, it seems like the authentication token should be persisted _immediately_ after authenticating so that it isn't lost if the app crashes.

**Comment by coffeejunk** [comment]
@voltapp Volt crashes for me when trying to load this public slack: https://makerskitchen.xyz

**Comment by s-uds** [comment]
> Same issue. Mine is a company-owned Slack.

Identical. It fetches channels/messages and crash at some point. Subsequent startup does not fix the issue.

**Comment by voltapp2** [comment]
Can you please try 0.53?

https://volt.ws/mac

**Comment by coffeejunk** [comment]
@voltapp still crashes for me. the authentication is also not persisted.

**Comment by s-uds** [comment]
Same here, is there anyway we can get a crash report of some sort?

**Comment by voltapp2** [comment]
@coffeejunk, @s-uds and everyone else :)

I found the reason of the crash: an out of bounds array access. V was catching it and silently exiting in prod build.

It was fixed in 0.54. Please restart Volt, it will automatically update to 0.54.

**Comment by coffeejunk** [comment]
@voltapp updated and it crashed again

**Comment by voltapp2** [comment]
@coffeejunk can you please open Terminal, run
`/Applications/Volt.app/Contents/MacOS/volt` and email me the output?

support@volt.ws

Thanks

**Comment by coffeejunk** [comment]
@voltapp hm now it's not crashing anymore 🤷‍♂️

**Comment by voltapp2** [comment]
Observer's paradox :)

I couldn't make it crash once.

**Comment by AdamVig** [comment]
Still crashes, but apparently a crash report was sent, so that's cool. Authentication is still not persisted. Auto-update does not seem to work, or maybe I'm using it wrong.

**Comment by voltapp2** [comment]
Adam, it’s really strange that authentication is not persisted and auto-updates don’t work.

Can you please run this in Terminal:

/Applications/Volt.app/Contents/MacOS/volt

and send me the output to support@volt.ws  ?

Thanks

**Comment by s-uds** [comment]
Still crashes (using 0.59 on macOS 10.14.1)

```c
2019-02-19 18:31:04.337 volt[12715:1066054] ch 99 / 100
Error decode for typ IMData : 
2019-02-19 18:31:04.341 volt[12715:1066054] im 0 / 31
Stack trace:
8!!
0   volt                                0x0000000100042430 f_359 + 48
1   libsystem_platform.dylib            0x00007fff7e872b3d _sigtramp + 29
2   ???                                 0x0000000032aaaba2 0x0 + 850045858
3   volt                                0x00000001000299c1 f_285 + 129
4   volt                                0x000000010001591e f_179 + 286
5   libsystem_pthread.dylib             0x00007fff7e87b339 _pthread_body + 126
6   libsystem_pthread.dylib             0x00007fff7e87e2a7 _pthread_start + 70
7   libsystem_pthread.dylib             0x00007fff7e87a445 thread_start + 13
2019-02-19 18:31:04.347 volt[12715:1066054] reopen_volt()
```

**Comment by voltapp2** [comment]
@s-uds that's a separate bug with occasional broken data coming from the server (most likely due to bad networking code).

I'll fix this today.

**Comment by voltapp2** [comment]
@s-uds could you post the entire log? Thanks
You can send it to support@volt.ws if you want

**Comment by s-uds** [comment]
Sure, here it is:

```c
/Applications/Volt.app/Contents/MacOS/volt
2019-02-19 18:29:21.792 volt[12715:1065913] starting volt v0.59 ...
2019-02-19 18:29:21.792 volt[12715:1065913] move volt to apps
2019-02-19 18:29:21.798 volt[12715:1065913] path app="/Applications/Volt.app"
2019-02-19 18:29:21.836 volt[12715:1065913] BUNDLE PATH /Applications/Volt.app
2019-02-19 18:29:21.836 volt[12715:1065913] Creating main NSWindow...
2019-02-19 18:29:21.887 volt[12715:1065913] focus1
2019-02-19 18:29:21.888 volt[12715:1065913] focus2
2019-02-19 18:29:21.889 volt[12715:1065913] Main NSWindow OK
2019-02-19 18:29:21.893 volt[12715:1065913] BUNDLE PATH /Applications/Volt.app
2019-02-19 18:29:22.305 volt[12715:1065913] latest version="0.59" curren version="0.59"
2019-02-19 18:29:22.305 volt[12715:1065913] BUNDLE PATH /Applications/Volt.app
2019-02-19 18:29:22.305 volt[12715:1065934] fetching profiles...
2019-02-19 18:29:22.310 volt[12715:1065934] got 0 profiles
2019-02-19 18:29:24.435 volt[12715:1065913] BUNDLE PATH /Applications/Volt.app
2019-02-19 18:29:24.435 volt[12715:1065913] Creating main NSWindow...
2019-02-19 18:29:24.441 volt[12715:1065913] focus1
2019-02-19 18:29:24.442 volt[12715:1065913] focus2
2019-02-19 18:29:24.444 volt[12715:1065913] Main NSWindow OK
2019-02-19 18:30:00.585 volt[12715:1066054] c create slack
2019-02-19 18:30:01.255 volt[12715:1066054] slack f OK
2019-02-19 18:30:01.476 volt[12715:1066054] c create slack - OK
2019-02-19 18:30:01.476 volt[12715:1066054] slack fetching chats...
2019-02-19 18:30:01.908 volt[12715:1066054] ch 0 / 100
2019-02-19 18:30:01.909 volt[12715:1066054] ch 1 / 100
2019-02-19 18:30:01.909 volt[12715:1066054] ch 2 / 100
2019-02-19 18:30:01.909 volt[12715:1066054] ch 3 / 100
2019-02-19 18:30:01.909 volt[12715:1066054] ch 4 / 100
2019-02-19 18:30:01.910 volt[12715:1066054] fetching msgs for CEAAKDPT2
2019-02-19 18:30:11.646 volt[12715:1066054] ch 5 / 100
2019-02-19 18:30:11.646 volt[12715:1066054] ch 6 / 100
2019-02-19 18:30:11.651 volt[12715:1066054] ch 7 / 100
2019-02-19 18:30:11.651 volt[12715:1066054] ch 8 / 100
2019-02-19 18:30:11.651 volt[12715:1066054] ch 9 / 100
2019-02-19 18:30:11.651 volt[12715:1066054] ch 10 / 100
2019-02-19 18:30:11.652 volt[12715:1066054] ch 11 / 100
2019-02-19 18:30:11.652 volt[12715:1066054] ch 12 / 100
2019-02-19 18:30:11.652 volt[12715:1066054] ch 13 / 100
2019-02-19 18:30:11.652 volt[12715:1066054] ch 14 / 100
2019-02-19 18:30:11.652 volt[12715:1066054] ch 15 / 100
2019-02-19 18:30:11.652 volt[12715:1066054] ch 16 / 100
2019-02-19 18:30:11.653 volt[12715:1066054] ch 17 / 100
2019-02-19 18:30:11.653 volt[12715:1066054] ch 18 / 100
2019-02-19 18:30:11.653 volt[12715:1066054] fetching msgs for CEAEH0GG1
2019-02-19 18:30:40.886 volt[12715:1066054] ch 19 / 100
2019-02-19 18:30:40.891 volt[12715:1066054] fetching msgs for CEAEH0QMP
2019-02-19 18:30:49.985 volt[12715:1066054] ch 20 / 100
2019-02-19 18:30:49.985 volt[12715:1066054] ch 21 / 100
2019-02-19 18:30:49.990 volt[12715:1066054] ch 22 / 100
2019-02-19 18:30:49.991 volt[12715:1066054] ch 23 / 100
2019-02-19 18:30:49.991 volt[12715:1066054] ch 24 / 100
2019-02-19 18:30:49.991 volt[12715:1066054] ch 25 / 100
2019-02-19 18:30:49.991 volt[12715:1066054] fetching msgs for CEAGQ63KL
2019-02-19 18:30:55.278 volt[12715:1066054] ch 26 / 100
2019-02-19 18:30:55.278 volt[12715:1066054] fetching msgs for CEANKST45
2019-02-19 18:30:55.512 volt[12715:1066054] ch 27 / 100
2019-02-19 18:30:55.512 volt[12715:1066054] ch 28 / 100
2019-02-19 18:30:55.518 volt[12715:1066054] ch 29 / 100
2019-02-19 18:30:55.518 volt[12715:1066054] ch 30 / 100
2019-02-19 18:30:55.518 volt[12715:1066054] fetching msgs for CEAT9USSY
Error decode for typ SlackUserInfo : 
Error decode for typ SlackUserInfo : 
Error decode for typ SlackUserInfo : 
Error decode for typ SlackUserInfo : 
Error decode for typ SlackUserInfo : 
2019-02-19 18:31:04.282 volt[12715:1066054] ch 31 / 100
2019-02-19 18:31:04.283 volt[12715:1066054] ch 32 / 100
2019-02-19 18:31:04.287 volt[12715:1066054] ch 33 / 100
2019-02-19 18:31:04.288 volt[12715:1066054] ch 34 / 100
2019-02-19 18:31:04.288 volt[12715:1066054] ch 35 / 100
2019-02-19 18:31:04.288 volt[12715:1066054] ch 36 / 100
2019-02-19 18:31:04.289 volt[12715:1066054] ch 37 / 100
2019-02-19 18:31:04.289 volt[12715:1066054] ch 38 / 100
2019-02-19 18:31:04.289 volt[12715:1066054] ch 39 / 100
2019-02-19 18:31:04.289 volt[12715:1066054] fetching msgs for CEATA001J
Error decode for typ HistoryData : 
2019-02-19 18:31:04.295 volt[12715:1066054] ch 40 / 100
2019-02-19 18:31:04.295 volt[12715:1066054] ch 41 / 100
2019-02-19 18:31:04.295 volt[12715:1066054] ch 42 / 100
2019-02-19 18:31:04.295 volt[12715:1066054] ch 43 / 100
2019-02-19 18:31:04.295 volt[12715:1066054] ch 44 / 100
2019-02-19 18:31:04.296 volt[12715:1066054] ch 45 / 100
2019-02-19 18:31:04.296 volt[12715:1066054] ch 46 / 100
2019-02-19 18:31:04.296 volt[12715:1066054] ch 47 / 100
2019-02-19 18:31:04.296 volt[12715:1066054] ch 48 / 100
2019-02-19 18:31:04.296 volt[12715:1066054] ch 49 / 100
2019-02-19 18:31:04.296 volt[12715:1066054] ch 50 / 100
2019-02-19 18:31:04.297 volt[12715:1066054] fetching msgs for CEAV8T6TD
Error decode for typ HistoryData : 
2019-02-19 18:31:04.303 volt[12715:1066054] ch 51 / 100
2019-02-19 18:31:04.303 volt[12715:1066054] ch 52 / 100
2019-02-19 18:31:04.303 volt[12715:1066054] fetching msgs for CEAV8TG03
Error decode for typ HistoryData : 
2019-02-19 18:31:04.309 volt[12715:1066054] ch 53 / 100
2019-02-19 18:31:04.310 volt[12715:1066054] ch 54 / 100
2019-02-19 18:31:04.310 volt[12715:1066054] ch 55 / 100
2019-02-19 18:31:04.310 volt[12715:1066054] ch 56 / 100
2019-02-19 18:31:04.310 volt[12715:1066054] ch 57 / 100
2019-02-19 18:31:04.310 volt[12715:1066054] ch 58 / 100
2019-02-19 18:31:04.311 volt[12715:1066054] ch 59 / 100
2019-02-19 18:31:04.311 volt[12715:1066054] fetching msgs for CEB2VFT26
Error decode for typ HistoryData : 
2019-02-19 18:31:04.317 volt[12715:1066054] ch 60 / 100
2019-02-19 18:31:04.317 volt[12715:1066054] ch 61 / 100
2019-02-19 18:31:04.317 volt[12715:1066054] ch 62 / 100
2019-02-19 18:31:04.317 volt[12715:1066054] ch 63 / 100
2019-02-19 18:31:04.318 volt[12715:1066054] ch 64 / 100
2019-02-19 18:31:04.318 volt[12715:1066054] ch 65 / 100
2019-02-19 18:31:04.318 volt[12715:1066054] fetching msgs for CEB2VG1A6
Error decode for typ HistoryData : 
2019-02-19 18:31:04.324 volt[12715:1066054] ch 66 / 100
2019-02-19 18:31:04.324 volt[12715:1066054] ch 67 / 100
2019-02-19 18:31:04.324 volt[12715:1066054] ch 68 / 100
2019-02-19 18:31:04.325 volt[12715:1066054] ch 69 / 100
2019-02-19 18:31:04.325 volt[12715:1066054] ch 70 / 100
2019-02-19 18:31:04.325 volt[12715:1066054] ch 71 / 100
2019-02-19 18:31:04.325 volt[12715:1066054] ch 72 / 100
2019-02-19 18:31:04.325 volt[12715:1066054] ch 73 / 100
2019-02-19 18:31:04.326 volt[12715:1066054] ch 74 / 100
2019-02-19 18:31:04.326 volt[12715:1066054] ch 75 / 100
2019-02-19 18:31:04.326 volt[12715:1066054] ch 76 / 100
2019-02-19 18:31:04.326 volt[12715:1066054] ch 77 / 100
2019-02-19 18:31:04.326 volt[12715:1066054] ch 78 / 100
2019-02-19 18:31:04.327 volt[12715:1066054] ch 79 / 100
2019-02-19 18:31:04.327 volt[12715:1066054] ch 80 / 100
2019-02-19 18:31:04.327 volt[12715:1066054] ch 81 / 100
2019-02-19 18:31:04.327 volt[12715:1066054] ch 82 / 100
2019-02-19 18:31:04.327 volt[12715:1066054] ch 83 / 100
2019-02-19 18:31:04.327 volt[12715:1066054] ch 84 / 100
2019-02-19 18:31:04.328 volt[12715:1066054] ch 85 / 100
2019-02-19 18:31:04.328 volt[12715:1066054] ch 86 / 100
2019-02-19 18:31:04.328 volt[12715:1066054] ch 87 / 100
2019-02-19 18:31:04.328 volt[12715:1066054] ch 88 / 100
2019-02-19 18:31:04.328 volt[12715:1066054] ch 89 / 100
2019-02-19 18:31:04.329 volt[12715:1066054] fetching msgs for CEC4C1YTZ
Error decode for typ HistoryData : 
2019-02-19 18:31:04.334 volt[12715:1066054] ch 90 / 100
2019-02-19 18:31:04.335 volt[12715:1066054] ch 91 / 100
2019-02-19 18:31:04.335 volt[12715:1066054] ch 92 / 100
2019-02-19 18:31:04.335 volt[12715:1066054] ch 93 / 100
2019-02-19 18:31:04.336 volt[12715:1066054] ch 94 / 100
2019-02-19 18:31:04.336 volt[12715:1066054] ch 95 / 100
2019-02-19 18:31:04.336 volt[12715:1066054] ch 96 / 100
2019-02-19 18:31:04.336 volt[12715:1066054] ch 97 / 100
2019-02-19 18:31:04.336 volt[12715:1066054] ch 98 / 100
2019-02-19 18:31:04.337 volt[12715:1066054] ch 99 / 100
Error decode for typ IMData : 
2019-02-19 18:31:04.341 volt[12715:1066054] im 0 / 31
Stack trace:
8!!
0   volt                                0x0000000100042430 f_359 + 48
1   libsystem_platform.dylib            0x00007fff7e872b3d _sigtramp + 29
2   ???                                 0x0000000032aaaba2 0x0 + 850045858
3   volt                                0x00000001000299c1 f_285 + 129
4   volt                                0x000000010001591e f_179 + 286
5   libsystem_pthread.dylib             0x00007fff7e87b339 _pthread_body + 126
6   libsystem_pthread.dylib             0x00007fff7e87e2a7 _pthread_start + 70
7   libsystem_pthread.dylib             0x00007fff7e87a445 thread_start + 13
2019-02-19 18:31:04.347 volt[12715:1066054] reopen_volt()
```

**Comment by voltapp2** [comment]
Perfect, thanks

**Comment by s-uds** [comment]
Any progress on this? Still crashes with 0.62

**Comment by voltapp2** [comment]
Still working on it. It’s tough because I can’t reproduce the crashes and have to debug with other users.

**Comment by voltapp2** [comment]
Can you please try 0.66?

**Comment by AdamVig** [comment]
Still crashed. I emailed the logs to you.

**Comment by s-uds** [comment]
0.67 still crashes, here are the logs:
[volt_0.67_crash_log.txt](https://github.com/voltapp/volt/files/2896710/volt_0.67_crash_log.txt)

**Comment by voltapp2** [comment]
Those of you still experiencing crashes, can you please try 0.68? It really should fix all the issues you had (judging by the logs).

**Comment by AdamVig** [comment]
I can verify that it no longer crashes in the same way, but I still have yet to see any actual messages in the app. Authentication is still not persisted across launches for me. This is unlikely to be relevant, but my company uses Google OAuth for Slack login.

**Comment by voltapp2** [comment]
> This is unlikely to be relevant, but my company uses Google OAuth for Slack login.

@AdamVig this must be it.

Can you please send me a screenshot of how authentication looks to support@volt.ws?

**Comment by drampelt** [comment]
I'm still having crashes with 0.69 (it autoupdated from 0.68), it said it sent the crash report

**Comment by voltapp2** [comment]
I know I've said it a million times, but please try 0.72.

(I wish I could reproduce this crash on my machine)

**Comment by s-uds** [comment]
Just tried 0.73 and it works! :)
Must've come from SAML / LDAP authentication for Slack login.

**Comment by drampelt** [comment]
Seems to be working a lot better now, it does finally load all messages and I can click on channels and I don't have to login again each time. Still occasionally crashing after being open for a few minutes or clicking on certain channels though

**Comment by AdamVig** [comment]
0.73 is working for me. This is my first time actually seeing messages in the app. Looks good! :tada:

**Comment by voltapp2** [comment]
@drampelt this crash you had was most likely fixed in 0.74.

**Comment by voltapp2** [comment]
All crashes have been finally fixed. Thank you very much for your help.
